src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -576,32 +528,15 @@ InstanceKlass* SystemDictionary::handle_parallel_loading(JavaThread* current,
       log_circularity_error(name, oldprobe);
       *throw_circularity_error = true;
       return nullptr;
-    } else {
+    } else if (must_wait_for_class_loading) {
       // Wait until the first thread has finished loading this class. Also wait until all the
       // threads trying to load its superclass have removed their placeholders.
       while (oldprobe != nullptr &&
              (oldprobe->instance_load_in_progress() || oldprobe->super_load_in_progress())) {
 
-        // We only get here if the application has released the
-        // classloader lock when another thread was in the middle of loading a
-        // superclass/superinterface for this class, and now
-        // this thread is also trying to load this class.
-        // To minimize surprises, the first thread that started to
-        // load a class should be the one to complete the loading
-        // with the classfile it initially expected.
-        // This logic has the current thread wait once it has done
-        // all the superclass/superinterface loading it can, until
-        // the original thread completes the class loading or fails
-        // If it completes we will use the resulting InstanceKlass
-        // which we will find below in the systemDictionary.
-
-        if (lockObject.is_null()) {
-          SystemDictionary_lock->wait();
-        } else if (EnableWaitForParallelLoad) {
-          double_lock_wait(current, lockObject);
-        } else {
-          return nullptr;
-        }
```

Comment:
Again unclear how this all disappears just because the flag is obsoleted.

Reply:
Only the bootclass loader waits on the SystemDictionary_lock for multiple threads now.  Not the non-parallel capable class loaders.

Reply:
Yes.